lib/ApplicationFeatures/LanguageFeature.cpp

```diff
@@ -176,13 +177,46 @@ void* LanguageFeature::prepareIcu(std::string const& binaryPath,
   return icuDataPtr;
 }
 
+void LanguageFeature::prepareTimeZoneData(std::string const& binaryPath,
+                                          std::string const& binaryExecutionPath,
+                                          std::string const& binaryName) {
+
+  std::string tz_path;
+  std::string test_exe = FileUtils::buildFilename(binaryExecutionPath, "tzdata");
+
+  if (FileUtils::isDirectory(test_exe)) {
+    FileUtils::makePathAbsolute(test_exe);
+    FileUtils::normalizePath(test_exe);
+    tz_path = test_exe;
+  } else {
+    std::string argv0 = FileUtils::buildFilename(binaryExecutionPath, binaryName);
+    std::string path = TRI_LocateInstallDirectory(argv0.c_str(), binaryPath.c_str());
+    path = FileUtils::buildFilename(path, ICU_DESTINATION_DIRECTORY, "tzdata");
+    FileUtils::makePathAbsolute(path);
+    FileUtils::normalizePath(path);
+    tz_path = path;
+  }
+
+  if (FileUtils::isDirectory(tz_path)) {
+    LOG_TOPIC("fffff", INFO, arangodb::Logger::STARTUP) << "using timezone data from " << tz_path;
+    date::set_install(tz_path);
+    date::reload_tzdb();
+  } else {
+    LOG_TOPIC("fffff", FATAL, arangodb::Logger::FIXME)
```

Comment:
For info, the log id strings need to be unique throughout the program. `fffff` is used twice here, so it won't pass our quick tests for log id uniqueness.
```suggestion
    LOG_TOPIC("67bdc", FATAL, arangodb::Logger::STARTUP)
```